plugin/pkg/admission/broker/authsarcheck/admission_test.go

```diff
@@ -0,0 +1,189 @@
+	"k8s.io/apiserver/pkg/authentication/user"
+
+	kubeinformers "k8s.io/client-go/informers"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	kubefake "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/pkg/api/v1"
+	authorizationapi "k8s.io/client-go/pkg/apis/authorization/v1"
+	core "k8s.io/client-go/testing"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	scadmission "github.com/kubernetes-incubator/service-catalog/pkg/apiserver/admission"
+)
+
+// newHandlerForTest returns a configured handler for testing.
+func newHandlerForTest(kubeClient kubeclientset.Interface) (admission.Interface, kubeinformers.SharedInformerFactory, error) {
+	kf := kubeinformers.NewSharedInformerFactory(kubeClient, 5*time.Minute)
+	handler, err := NewSARCheck()
+	if err != nil {
+		return nil, kf, err
+	}
+	pluginInitializer := scadmission.NewPluginInitializer(nil, nil, kubeClient, kf)
+	pluginInitializer.Initialize(handler)
+	err = admission.Validate(handler)
+	return handler, kf, err
+}
+
+// newMockKubeClientForTest creates a mock kubernetes client that is configured
+// to allow any SAR creations.
+func newMockKubeClientForTest(userInfo *user.DefaultInfo) *kubefake.Clientset {
+	mockClient := &kubefake.Clientset{}
+	allowed := true
+	if userInfo.GetName() == "system:serviceaccount:test-ns:forbidden" {
+		allowed = false
+	}
+	mockClient.AddReactor("create", "subjectaccessreviews", func(action core.Action) (bool, runtime.Object, error) {
+		mysar := &authorizationapi.SubjectAccessReview{
+			Status: authorizationapi.SubjectAccessReviewStatus{
+				Allowed: allowed,
+				Reason:  "seemed friendly enough",
+			},
+		}
+		return true, mysar, nil
+	})
+	return mockClient
+}
+
+// TestAdmissionBroker tests Admit to ensure that the result from the SAR check
+// is properly checked.
+func TestAdmissionBroker(t *testing.T) {
+	// Anonymous struct fields:
+	// name: short description of the testing
+	// broker: a fake broker object
+	// allowed: flag for whether or not the broker should be admitted
+	cases := []struct {
+		name     string
+		broker   *servicecatalog.ServiceBroker
+		userInfo *user.DefaultInfo
+		allowed  bool
+	}{
+		{
+			name: "broker with no auth",
+			broker: &servicecatalog.ServiceBroker{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-broker",
+				},
+				Spec: servicecatalog.ServiceBrokerSpec{
+					URL: "http://example.com",
+				},
+			},
+			userInfo: &user.DefaultInfo{
+				Name:   "system:serviceaccount:test-ns:catalog",
+				Groups: []string{"system:serviceaccount", "system:serviceaccounts:test-ns"},
+			},
+			allowed: true,
+		},
+		{
+			name: "broker with basic auth, user authenticated",
+			broker: &servicecatalog.ServiceBroker{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-broker",
+				},
+				Spec: servicecatalog.ServiceBrokerSpec{
+					URL: "http://example.com",
+					AuthInfo: &servicecatalog.ServiceBrokerAuthInfo{
+						Basic: &servicecatalog.BasicAuthConfig{
```

Comment:
Can you add a test for the same use case but with deprecated `serviceBroker.Spec.AuthInfo.BasicAuthSecret` field set instead?

Reply:
added!